### PR TITLE
[C-2082] Fix busted profile edits

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description

Couple things:
* Pull to refresh totally breaks profile updates. In commit 1, make it so that we never override something that's in the confirmer. The confirmer finishes by refetching and caching the new object, so we should avoid doing this while confirming. Mobile web (prior to reloaded) had a solve for this that I've deleted since it's pretty brittle
* Fix profile/cover photo edits by allowing the local updates to be used in fastimage, etc.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Local simulator & web browser for commit 1

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

File: packages/mobile/src/components/image/UserCoverImage.tsx [hunk 1]
```diff
@@ -31,12 +31,21 @@ type CoverImageUser = Nullable<
...
+  const contentNodeImage = useContentNodeImage({
     cid,
     user,
     size: WidthSizes.SIZE_640,
     fallbackImageSource: imageCoverPhotoBlank
   })
+
+  if (user.updatedCoverPhoto) {
+    return {
+      source: { uri: user.updatedCoverPhoto.url },
```
Comment (dylanjeffers) [inline packages/mobile/src/components/image/UserCoverImage.tsx:43]: That's fun

File: packages/mobile/src/components/image/UserImage.tsx [hunk 2]
```diff
@@ -19,12 +19,20 @@ type UseUserImageOptions = {
...
 
-  return useContentNodeImage({
+  const contentNodeImage = useContentNodeImage({
     cid,
     size,
     user,
     fallbackImageSource: profilePicEmpty
   })
+
+  if (user.updatedProfilePicture) {
```
Comment (dylanjeffers) [inline packages/mobile/src/components/image/UserImage.tsx:29]: Would we want to add this to the core image hook or just a mobile only thing for now?

File: packages/web/src/common/store/cache/sagas.js [hunk 3]
```diff
@@ -98,15 +99,21 @@ export function* retrieve({
     call(getEntriesTimestamp, uniqueIds)
   ])
 
+  const confirmer = yield select(getConfirmCalls)
   // Figure out which IDs we need to retrive from source
   const idsToFetch = []
   uniqueIds.forEach((id) => {
-    if (
+    let isConfirming = false
+    if (cachedEntries?.[id]?.[idField]) {
```
Comment (dylanjeffers) [inline packages/web/src/common/store/cache/sagas.js:107]: Awesome adds. Isn't the cache add step check if confiming, and if so don't call add_successful?

File: packages/web/src/common/store/cache/users/sagas.js [hunk 4]
```diff
@@ -213,17 +213,17 @@ function* watchFetchProfilePicture() {
...
-                [size]: url
-              }
-            }
             yield put(
               cacheActions.update(Kind.USERS, [
-                { id: userId, metadata: userWithProfilePicture }
+                {
+                  id: userId,
+                  metadata: {
+                    _profile_picture_sizes: {
```
Comment (dylanjeffers) [inline packages/web/src/common/store/cache/users/sagas.js:221]: Nice fix, so basically don't merge whole object?

**Review by dylanjeffers — APPROVED** [review]
Awesome stuff

File: packages/mobile/src/components/image/UserImage.tsx (hunk 2 above)
Comment (raymondjacobson) [reply packages/mobile/src/components/image/UserImage.tsx:29]: It's handled separately on web for now, so this at least brings parity. Would be good to move deeper into common though

File: packages/web/src/common/store/cache/users/sagas.js (hunk 4 above)
Comment (raymondjacobson) [reply packages/web/src/common/store/cache/users/sagas.js:221]: yeah same as your fix

File: packages/web/src/common/store/cache/sagas.js (hunk 3 above)
Comment (raymondjacobson) [reply packages/web/src/common/store/cache/sagas.js:107]: You're right, we already had code to do this. It seems like pull to refresh explicitly passes through an arg to blow away the current entry in the cache. Maybe we can huddle abt this, not quite sure what the reasoning for that was

Comment (raymondjacobson) [reply packages/web/src/common/store/cache/sagas.js:107]: Ok I removed the code I added here and also dropped the arg in pull to refresh and I see the same behavior as with my fix. Mine might be better b/c we lose a network hop, but it's definitely a riskier change so will merge without.

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

File: packages/web/src/common/store/cache/sagas.js (hunk 3 above)
Comment (dylanjeffers) [reply packages/web/src/common/store/cache/sagas.js]: Whatever you think is best, this can be something we discuss during irl too

**Pull request merged**